### ISSUE TITLE
LRP with fine-tuned ResNet101 - attribute values are nans

### ISSUE THREAD

**Issue by annasli378** [pytorch/captum#1138]
## ❓ Questions and Help

Hi, 
I am having troubles with LRP and my model ResNet101. As input i give metrod my image as a tensor and as target i used tensor with my target value as well as only integers. I used different methods (IG, Occlusion, Saliency) and they work well. 

I even tried using random values and zeros as my input_tensor, with the same result.

My code:

```
lrp = LRP(model)


output = model(image)
_, predicted = torch.max(output.data, 1)    
original_image = np.transpose((image.squeeze().numpy() / 2) + 0.5, (1, 2, 0))
            
attr_lrp  = lrp.attribute(input_tensor, target=predicted)  

```

What i get as attr_lrp  is a tensor filled with nan values  with size as my input : torch.Size([1, 3, 224, 224]).
I did not found anything similar in help or documentation.

Do anyone know how can I obtain attributions from LRP or what could possibly couse this behaviour?

**Comment by craymichael** [comment]
Apologies for the delay.

NaNs in LRP usually come from unstable relevance propagation through one or more layers, often because a denominator is near zero or because a module does not have an appropriate LRP rule.

Things I would check first:

```python
model.eval()
```

Use an integer target rather than a tensor target:

```python
target = int(predicted.item())
attr_lrp = lrp.attribute(image, target=target)
```

Then try assigning a more stable epsilon rule to sensitive layers:

```python
from captum.attr._utils.lrp_rules import EpsilonRule

for module in model.modules():
    if isinstance(module, (torch.nn.Conv2d, torch.nn.Linear, torch.nn.BatchNorm2d)):
        module.rule = EpsilonRule(epsilon=1e-6)
```

Also make sure the input normalization matches the fine-tuned model. If other methods work but LRP returns NaNs, the model may contain an operation / block for which the default LRP rule is numerically unstable. In that case, checking `return_convergence_delta=True` and assigning explicit rules layer-by-layer can help isolate the problematic module.

This should answer the question, so I'll close this for now. Please feel free to reply with any follow-up questions.
